File: Fines2Model3/DAO/Planificacion.cs


namespace SqlOrganize.Sql.Fines2Model3
{
    public static class PlanificacionDAO
    {
        public static Planificacion? PlanificacionSiguiente__By_IdPlan_Anio_Semestre(object idPlan, object anio, object semestre)
        {
            (string anio_, string semestre_) = Planificacion.AnioSemestreSiguiente(anio, semestre);

            string sql = @"SELECT id FROM planificacion 
                WHERE anio = @Anio
                AND semestre = @Semestre
                AND plan = @IdPlan;
            ";

            return Context.db.CacheSql().QueryIds<Planificacion>(sql, new { Anio = anio, Semestre = semestre, IdPlan = idPlan}).FirstOrDefault();

        }
    }
}

File: Data/Planificacion_.php
<?php

namespace Fines2;

use \Fines2\Planificacion;
use SqlOrganize\Sql\DbMy;

class Planificacion_ extends Planificacion
{
    public function getTramo(): string {
        return ($this->anio ?? "?") . "°" . ($this->semestre ?? "?") . "C";
    }

    public function getLabel(): string {
        return ($this->plan_?->orientacion ?? "?") . " " 
        . ($this->plan_?->resolucion  ?? "?") . " "
        . ($this->anio  ?? "?") . "/"
        . ($this->semestre  ?? "?");
    }

    public static function planificaciones(): array{
        $db = DbMy::getInstance();

        $dataProvider = $db->CreateDataProvider();

        $sql = "SELECT planificacion.id
                FROM planificacion 
                INNER JOIN plan ON planificacion.plan = plan.id 
                ORDER BY plan.resolucion, plan.orientacion, planificacion.anio, planificacion.semestre";

        $entities = $dataProvider->fetchEntitiesBySqlId("planificacion", $sql);
        return $entities;
    }




}

